homeassistant/components/airtouch4/__init__.py

```diff
@@ -0,0 +1,104 @@
+"""The AirTouch4 integration."""
+import logging
+
+from airtouch4pyapi import AirTouch
+
+from homeassistant.components.climate import SCAN_INTERVAL
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["climate"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the AirTouch4 component."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up AirTouch4 from a config entry."""
+    host = entry.data[CONF_HOST]
+    airtouch = AirTouch(host)
+    try:
+        info = airtouch.GetAcs()
+        if not info:
+            raise ConfigEntryNotReady
+    except (OSError, ConnectionRefusedError, TimeoutError) as error:
+        raise ConfigEntryNotReady() from error
+    coordinator = AirtouchDataUpdateCoordinator(hass, airtouch)
+    await coordinator.async_refresh()
+    hass.data[DOMAIN][entry.entry_id] = {
+        "info": {
```

Comment:
I am a bit lost, where are you using this? It looks like the same code as defined in the DataUpdateCoordinator. Why not pull this data from the coordinator. (`coordinator.data`)